### PR TITLE
Fix cache read and write input param types being generated incorrectly

### LINKED ISSUE #1129: Typescript types of variable inputs for cache read and write functions are incorrect
### Describe the bug

When using the Cache API, the requested variables for the query in `Cache.read` or `Cache.write` correspond to the `$result` type (instead of `$input`), which gives an error when using Typescript

Printscreen of the error:

Screenshot 2023-07-11 at 16 21 38

### Severity

annoyance

### Steps to Reproduce the Bug

1. Make sure you have a query that requires variables to be passed through, and that you are using Typescript
2. Use one of the Cache API methods: `write` or `read`
3. When passing the variables to that function, you'll see a type error

### Reproduction

_No response_

### PR DESCRIPTION
**Pull request by SeppahBaws**
Fixes #1129 

The type definition of QueryInput used QueryValue, which resulted into it becoming `MyQuery$result`, instead of the expected `MyQuery$input`.

I'm not 100% sure how to write a proper test for this, suggestions welcome 🙂 

### To help everyone out, please make sure your PR does the following:

- [x] Update the first line to point to the ticket that this PR fixes
- [x] Add a message that clearly describes the fix
- [ ] If applicable, add a test that would fail without this fix
- [x] Make sure the unit and integration tests pass locally with `pnpm run tests` and `cd integration && pnpm run tests`
- [x] Includes a changeset if your fix affects the user with `pnpm changeset`

### CODE REVIEW

**Comment by AlecAivazis** [comment]
Thanks for getting this out! To write a test for this, you could just use the API in the e2e tests. we run a linter/typechecker on the project as part of the CI flow so it will catch type errors

**Comment by SeppahBaws** [comment]
Running into a very weird bug here while adding a test, will need some more digging to find the root cause 🧐 
It seems to be grabbing the `$result` type from the query above
image
image

File: packages/houdini/src/runtime/public/types.ts [hunk 1]
```diff
@@ -103,34 +103,61 @@ export type ListType<Def extends CacheTypeDef, Name extends ValidLists<Def>> = P
 // then typescript wasn't responding on every change... Having the duplication makes
 // it very responsive.
 
+/**
```
Comment (AlecAivazis) [inline packages/houdini/src/runtime/public/types.ts:106]: love this comment!

**Review by AlecAivazis — APPROVED** [review]
Nice and easy. Thanks again!

**Pull request merged**